### PR TITLE
[Tests-only] Add changelog step to makefile

### PR DESCRIPTION
**Pull request by micbar**
## Description
- Add Changelog generate to makefile

## Related Issue
- https://github.com/owncloud/core/pull/37197

## Motivation and Context
- Our CHANGELOG.md is only generated automatically on the master branch
- This make target could help to create a up-to date changelog anytime on a branch for testing.

## How Has This Been Tested?
- manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -358,6 +359,15 @@ dist-dir-qa: $(dist_dir)/qa/owncloud
 update-php-license-header:
 	php build/license.php
 
+.PHONY: changelog
+changelog:
```
Comment (phil-davis) [inline Makefile:363]: Do you want to also add to the help in the "Tools" section (after line 132)?
Or keep quiet about this target?

**Review by jnweiger — APPROVED** [review]
Excellent addition! Thanks!
That helps to simplify release scripts.

**Review by phil-davis — APPROVED** [review]
It works - I made a local test changelog file with a title that is too long and:
```
$ make changelog
======================================================================
WARNING: Please do NOT commit changes in the CHANGELOG.md file to git!
This file is autogenerated on the master branch.
Committing changes to this file will result in merge conflicts.
======================================================================
docker run --rm -v /home/phil/git/owncloud/core:/home/phil/git/owncloud/core -w /home/phil/git/owncloud/core toolhippie/calens >| CHANGELOG.md
file changelog/unreleased/37999: title is too long
Makefile:364: recipe for target 'changelog' failed
make: *** [changelog] Error 1
```

File: Makefile (hunk 1 above)
Comment (micbar) [reply Makefile:363]: good idea, will do

**Review by phil-davis — APPROVED** [review]

**Pull request merged**